pallets/frequency-tx-payment/src/lib.rs

```diff
@@ -440,18 +452,48 @@ where
 		len: usize,
 	) -> Result<(BalanceOf<T>, InitialPayment<T>), TransactionValidityError> {
 		let mut calls_weight_sum = Weight::zero();
+		let mut subsidized_calls_weight_sum = Weight::zero();
+
 		for call in calls {
 			let call_weight = T::CapacityCalls::get_stable_weight(call)
 				.ok_or(ChargeFrqTransactionPaymentError::CallIsNotCapacityEligible.into())?;
 			calls_weight_sum = calls_weight_sum.saturating_add(call_weight);
-		}
-		let fee = Pallet::<T>::compute_capacity_fee(len as u32, calls_weight_sum);
 
-		let fee = T::OnChargeCapacityTransaction::withdraw_fee(key, fee.into())?;
+			if self.call_is_adding_eligible_key_to_msa(call) {
+				subsidized_calls_weight_sum =
+					subsidized_calls_weight_sum.saturating_add(call_weight);
+			}
+		}
+		let capacity_fee = Pallet::<T>::compute_capacity_fee(len as u32, calls_weight_sum)
+			.saturating_sub(Self::subsidized_calls_reduction(len, subsidized_calls_weight_sum));
+		let fee = T::OnChargeCapacityTransaction::withdraw_fee(key, capacity_fee.into())?;
 
 		Ok((fee.into(), InitialPayment::Capacity))
 	}
 
+	// Reduction for subsidized calls.
+	fn subsidized_calls_reduction(len: usize, eligible_call_weight: Weight) -> BalanceOf<T> {
+		if eligible_call_weight.is_zero() {
+			0u32.into()
+		} else {
+			let reduction: Permill = Permill::from_percent(70u32);
```

Comment:
See e2e test for how much this reduces cost.

Reply:
nit: I think it might be helpful to have a comment about the weight reductions or percentages and such in here. 

Overall is it fair to say that this reduces the fee to around 0.3 or a bit more for subsidized?

Reply:
Yes, that's about right, somewhere between 25-30%